lighthouse-core/gather/gatherers/obscuring-elements.js

```diff
@@ -0,0 +1,115 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+const TraceProcessor = require('../../lib/tracehouse/trace-processor.js');
+const { getOuterHTMLSnippet } = require('../../lib/page-functions.js');
+
+/** @typedef {{element?: string, x?: number, y?:number, retList?: Array<any>}} MyData */
+/** @typedef {{xCoord: number, yCoord: number}} Point */
+
+/**
+ * @this {HTMLElement}
+ * @return {MyData}
+ */
+/* istanbul ignore next */
+function checkIntersection() {
+  const lcpElement = this.nodeType === document.ELEMENT_NODE ? this : this.parentElement; // eslint-disable-line no-undef
+  const lcpHTMLSnippet = lcpElement && getOuterHTMLSnippet(lcpElement);
+  const obscuringElements = [];
+  const seen = new Set();
+  /**
+   * @param {HTMLElement | Element} el 
+   */
+  function getChildrenSnippets(el) {
+    if (!el) return;
+    const snippet = getOuterHTMLSnippet(el);
+    seen.add(snippet);
+    for (const child of el.children) {
+      getChildrenSnippets(child);
+    }
+  }
+  lcpElement && getChildrenSnippets(lcpElement);
+  const boundingRect = lcpElement && lcpElement.getBoundingClientRect();
+  if (!boundingRect) return {};
+  const {x, y, width, height} = boundingRect;
+  /** @type {Array<Point>} */
+  const points = [];
+  for (let i = 0.1; i < 1.0; i += 0.2) {
+    for (let j = 0.1; j < 1.0; j += 0.2) {
+      const xCoord = x + Math.round(width * i);
+      const yCoord = y + Math.round(height * j);
+      points.push({xCoord, yCoord});
+    }
+  }
+
+  for (const {xCoord, yCoord} of points) {
+    for (const element of document.elementsFromPoint(xCoord, yCoord)) {
+      // @ts-ignore - put into scope via stringification
+      const snippet = getOuterHTMLSnippet(element); // eslint-disable-line no-undef
+      // if (snippet === lcpHTMLSnippet) break;
+      if (element === lcpElement) break;
+      if (seen.has(snippet))
+        continue;
+      seen.add(snippet);
+      const rect = element.getBoundingClientRect();
+      obscuringElements.push({x: rect.x, y: rect.y, width: rect.width, height: rect.height, snippet});
+    }
+  }
+
+  return {
+    element: lcpHTMLSnippet || '',
+    x,
+    y,
+    retList: obscuringElements,
+  };
+}
+
+class ObscuringElements extends Gatherer {  
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @param {LH.Gatherer.LoadData} loadData
+   * @return {Promise<LH.Artifacts['ObscuringElements'] | null>}
+   */
+  async afterPass(passContext, loadData) {
```

Comment:
talked to @Beytoven

let's try grabbing all the obscuring elements, sort by size, and for each: iterate against all the other elements and remove any that are children of the current elements.

what we should be left with is the largest element of each distinct part of the DOM (it's possible that through css trickery multiple "distinct" parts of the DOM could overlap with LCP)